File: app/src/main/java/com/yyxnb/arch/frag/TestFragment.java
package com.yyxnb.arch.frag;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.yyxnb.arch.R;
import com.yyxnb.arch.TestActivity;
import com.yyxnb.yyxarch.base.BaseFragment;
import com.yyxnb.yyxarch.interfaces.LayoutResId;
import com.yyxnb.yyxarch.interfaces.StatusBarColor;
import com.yyxnb.yyxarch.interfaces.StatusBarDarkTheme;
import com.yyxnb.yyxarch.interfaces.BarStyle;
import com.yyxnb.yyxarch.utils.ToastUtils;

import org.jetbrains.annotations.Nullable;

import butterknife.BindView;

/**
 * A simple {@link Fragment} subclass.
 */
@LayoutResId(value = R.layout.fragment_test)
@StatusBarColor(color = R.color.orange)
@StatusBarDarkTheme(value = BarStyle.LightContent)
public class TestFragment extends BaseFragment {


    @BindView(R.id.tvShow)
    TextView tvShow;
    @BindView(R.id.button)
    Button button;
    @BindView(R.id.button1)
    Button button1;
    @BindView(R.id.button2)
    Button button2;
    @BindView(R.id.button3)
    Button button3;
    @BindView(R.id.button4)
    Button button4;



    @Override
    public void initView(Bundle savedInstanceState) {

//        tvShow = findViewById(R.id.tvShow);
//        button = findViewById(R.id.button);
//        button1 = findViewById(R.id.button1);
//        button2 = findViewById(R.id.button2);
//        button3 = findViewById(R.id.button3);
//        button4 = findViewById(R.id.button4);

    }

    @Override
    public void initViewData() {
        super.initViewData();

        tvShow.setText(getDebugTag());

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(TestActivity.class);
            }
        });


        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                startFragment(new TestFragment());

            }
        });

        button2.setOnClickListener(v -> {

            startFragment(new TwoFragment());
        });

        button3.setOnClickListener(v -> {
            startFragment(new ContentViewFragment());
        });

        button4.setOnClickListener(v -> {

            finish();
        });

    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == 0x111 && resultCode == 0x110) {
            ToastUtils.INSTANCE.normal("onActivityResult  " + data.getExtras().getString("msg", "null"));
        }
    }

    public static TestFragment newInstance() {

        Bundle args = new Bundle();

        TestFragment fragment = new TestFragment();
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/yyxnb/arch/frag/TestFragment.kt
package com.yyxnb.arch.frag


import android.graphics.Color
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.yyxnb.arch.R
import com.yyxnb.yyxarch.base.BaseFragment
import com.yyxnb.yyxarch.utils.BarStyle
import com.yyxnb.yyxarch.utils.ToastUtils
import com.yyxnb.yyxarch.utils.log.LogUtils
import kotlinx.android.synthetic.main.fragment_test.*


/**
 * A simple [Fragment] subclass.
 */
class TestFragment : BaseFragment() {


    private var value: String? = null

    override fun initLayoutResID(): Int {
        return R.layout.fragment_test
    }

    override fun initView(savedInstanceState: Bundle?) {



        value = arguments!!.getString("value")
    }

    override fun initVariables(bundle: Bundle) {
        super.initVariables(bundle)

        value = bundle.getString("value")
    }

    override fun initViewData() {
        super.initViewData()

        //        tvShow.setText(getDebugTag());
        tvShow.text = value

        LogUtils.w("--- " + value!!)

        button.setOnClickListener { startActivityRootFragment(TestFragment(), true) }


        button1.setOnClickListener { startFragment(TestFragment()) }

        button2.setOnClickListener { v ->

            startFragment(TwoFragment(), 0x111)
        }

        button3.setOnClickListener { v ->

            val navigationFragment = getNavigationFragment()
            navigationFragment?.popToRootFragment()
        }

        button4.setOnClickListener { v ->

            finish()
        }

    }

    override fun preferredStatusBarStyle(): BarStyle {
        return BarStyle.DarkContent
    }

    override fun preferredStatusBarColor(): Int {
        return Color.TRANSPARENT
    }

    override fun onFragmentResult(requestCode: Int, resultCode: Int, result: Bundle?) {
        super.onFragmentResult(requestCode, resultCode, result)
        if (requestCode == 0x111 && resultCode == 0x110) {
            ToastUtils.normal("0x111 " + result!!.getString("msg", "null"))
        }
    }

    companion object {

        fun newInstance(value: String): TestFragment {

            val args = Bundle()
            args.putString("value", value)
            val fragment = TestFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
